starknet-foundry/crates/forge/src/scarb.rs

```diff
@@ -140,7 +140,14 @@ pub fn dependencies_for_package(
 
     let target_name = compilation_unit.target.name.clone();
 
-    Ok((base_path, dependencies, target_name))
+    let corelib = compilation_unit
+        .components
+        .iter()
+        .find(|du| du.source_path.to_string().contains("core/src"))
```

Comment:
Just to make sure, isn't there a more direct way of finding corelib through scarb?

Reply:
Not sure, but I think this way is the best - we will have appropriate version of corelib every time + Marek told me that metadata call can sometimes change the content of the corelib (resolving etc.)